File: server/my-family-ws/src/main/java/com/bytepushers/family/model/Role.java
package com.bytepushers.family.model;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "roles")
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String name;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "role_id")
    private List<Permission> permissions;

    // Default Constructors
    public Role(){

    }

    public Role(Integer id, String name, List<Permission> permissions) {
        this.id = id;
        this.name = name;
        this.permissions = permissions;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Permission> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<Permission> permissions) {
        this.permissions = permissions;
    }

    @Override
    public String toString() {
        return "Role{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", permissions=" + permissions +
                '}';
    }
}

File: client/my-family-app/src/app/models/role.ts
import { Permission } from './permission';  // Import Permission class

export class Role {
  #name: string;
  #permissions: Permission[];

  constructor(name: string, permissions: Permission[]) {
    this.#name = name;
    this.#permissions = permissions;
  }

  // Getters
  public getName(): string {
    return this.#name;
  }

  public get name(): string {
    return this.#name;
  }

  public getPermissions(): Permission[] {
    return this.#permissions;
  }

  public get permissions(): Permission[] {
    return this.#permissions;
  }

  // toString method for debugging
  public toString(): string {
    return `"role": {
              "name": "${this.#name}",
              "permissions": [${this.#permissions.map(permission => permission.toString()).join(', ')}]
            }`;
  }
}
